src/OperationResponse.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+// Simple case
+$op = $sampleApi->longRunningRpc();
+// ... do stuff ...
+$op->pollUntilComplete();
+$result = $op->getResult();
+
+// Save handle and resume
+$opName = $sampleApi->longRunningRpc()->getName();
+// ... do stuff ...
+$op = $sampleApi->getOperationsApi()->getOperation($opName);
+$op->pollUntilComplete();
+$result = $op->getResult();
+
+// Polling loop
+$op = $sampleApi->longRunningRpc();
+while (!$op->isDone()) {
+    // ... do stuff ...
+    $op->refresh();
+}
+$result = $op->getResult();
+
+// Polling loop with promise
+$op = $sampleApi->longRunningRpc();
+$op->promise()->then($funcToHandleResponse, $funcToHandleError);
+while (true) {
+    // ... do stuff ...
+    $op->refresh();
+}
+
+// Sample using operation proto object
+$op = $sampleApi->longRunningRpc();
+// ... do stuff ...
+$op->pollUntilComplete();
+$opProto = $op->getProtoResponse();
+// ... use opProto ...
+
+
+//////////////////////////////////////////////////////////////////
+// Samples when returning an array with flag for checking success
+list($success, $response) = $sampleApi->longRunningRpc()->pollUntilComplete();
+if ($success) {
+    // handle success
+} else {
+    // handle failure
+}
+
+list($statusCode, $response) = $sampleApi->longRunningRpc()->pollUntilComplete();
+if ($statusCode == Google\Rpc\Code::OK) {
+    // handle success
+} else {
+    // handle failure
+}
+
+
+class OperationResponse
+{
+    private $operationProto;
+    private $operationsApi;
+
+    // OPTIONAL: support promises using a deferred object using
+    // https://github.com/reactphp/promise#deferred or similar
+    private $deferred;
+
+    public function isDone()
+    {
+        return $this->operationProto->getDone();
+    }
+
+    public function getName()
+    {
+        return $this->operationProto->getName();
+    }
+
+    public function pollUntilComplete($pollSettings = [])
+    {
+        while (!$this->isDone()) {
+            // TODO: use poll settings
+            sleep(1);
+            $this->refresh();
+        }
```

Comment:
Very possibly! Originally it was tricky because I had getResult() throwing an exception on errors. If we go with the suggestion from @jdpedrie below, then it could return getResult(), which will be null in the case of errors. It could also return a bool indicating success or error.